### PR TITLE
add cookies commands

### PR DESCRIPTION
**Pull request by ryanlerch**
this adds two new commands for cookies:

* !cookie give -- which does the same thing as name++
* !cookie count -- which gives a count of cookies for a user

### CODE REVIEW

**Review by abompard — CHANGES_REQUESTED** [review]

File: fedora/cookie.py [hunk 1]
```diff
@@ -56,6 +57,29 @@ def _get_username(self, evt: MessageEvent) -> str | None:
             return
         return match.group(1)
 
+    async def _get_cookie_totals(self, username):
+        dbq = "SELECT SUM(value) FROM cookies WHERE to_user = $1"
+        total = await self.plugin.database.fetchval(dbq, username)
+        by_release = {}
+
+        dbq = (
+            "SELECT release, count(release) AS count "
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply fedora/cookie.py:66]: fixed -- also expanded the test on this for better testing and with values other than 1

File: fedora/cookie.py [hunk 2]
```diff
@@ -83,6 +107,48 @@ async def give(self, evt: MessageEvent, username: str) -> None:
...
+            await self.give(evt, username)
+        except InfoGatherError as e:
+            await evt.respond(e.message)
+            return
+
+    @cookie.subcommand(name="count", help="Return the cookie count for a user")
+    @command.argument("username", required=True)
+    async def cookie_count(self, evt: MessageEvent, username: str) -> None:
+        user = await get_fasuser(username or evt.sender, evt, self.plugin.fasjsonclient)
+        self.plugin.log.warn(user)
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply fedora/cookie.py:137]: fixed

File: fedora/cookie.py [hunk 3]
```diff
@@ -56,6 +57,29 @@ def _get_username(self, evt: MessageEvent) -> str | None:
             return
         return match.group(1)
 
+    async def _get_cookie_totals(self, username):
+        dbq = "SELECT SUM(value) FROM cookies WHERE to_user = $1"
```
[the remark this answers is not in the document]
Comment (abompard) [reply fedora/cookie.py:61]: I suggest adding to `db.py`:

```diff
@@ -38,3 +38,12 @@ async def upgrade_v2(conn: Connection) -> None:
         CREATE INDEX idx_cookies_to_user_release ON cookies (to_user, release);
     """
     )
+
+
+@upgrade_table.register(description="Add index on cookie.to_user")
+async def upgrade_v3(conn: Connection) -> None:
+    await conn.execute(
+        """
+        CREATE INDEX idx_cookies_to_user ON cookies (to_user);
+    """
+    )
```

[the remark this answers is not in the document]
Comment (ryanlerch) [reply fedora/cookie.py:61]: thanks! added

**Comment by ryanlerch** [comment]
just waiting on the checks to pass in CI, and then i will merge, and merge into staging

**Review by abompard — APPROVED** [review]

**Pull request merged**